src/provider/CoinbaseWalletProvider.ts

```diff
@@ -347,7 +351,9 @@ export class CoinbaseWalletProvider
   }
 
   public close() {
-    void this.initializeRelay().then(relay => relay.resetAndReload());
+    void this.initializeRelay().then(relay =>
+      this._headlessMode ? relay.reset() : relay.resetAndReload()
+    );
```

Comment:
yup!

Reply:
actually, i'm not so sure – does changing the return type from `void` to `Promise<void>` constitute a breaking change?

lint is [failing](https://github.com/RyanTimesTen/coinbase-wallet-sdk/commit/79ef514bfaff47b1efa6a6ba703195c37bdfcedf#annotation_3336979697) and wants us to `await` the call to `close`, which we can do by updating`disconnect` to return `Promise<void>` as well. but then, coinbase-wallet-sdk consumers will potentially run into the same issue.

one option is to introduce new methods that return `Promise<void>`, and leave the existing ones as-is. what are your thoughts @ljharb ?

Reply:
I don't think it does - someone currently ignoring the return value can continue to do so, but someone awaiting it will now have their bug fixed, by properly waiting for the task to complete.

iow, it seems like a bugfix if anything

Reply:
good point

Reply:
updated, thank you!